tools/verifyio/verifyio.py

```diff
@@ -62,11 +80,17 @@ def verify_pair_proper_synchronization(n1, n2, vio):
         v1 = vio.G.next_hb_node(n1, ["fsync", "close", "fclose"], rank)
         v2 = n2
     elif vio.semantics == "Session":
-        v1 = vio.G.next_po_node(n1, ["close", "fclose", "fsync"])
-        v2 = vio.G.prev_po_node(n2, ["open",  "fopen",  "fsync"])
+        if vio.G is None:
+            v1, v2, next_sync_index = find_next_prev_sync(n1, n2, all_nodes, ["close", "fclose", "fsync"], ["open",  "fopen",  "fsync"])
```

Comment:
no need to do this. once you have v1 and v2 (from next_po_node/prev_po_node), you could get the index using `index = self.G.nodes[v1.graph_key()]['index']`